test/test_networking.py

```diff
@@ -889,6 +908,23 @@ def mock_read(*args, **kwargs):
 
         assert exc_info.type is expected
 
+    def test_close(self, handler, monkeypatch):
+        rh = handler()
+        session = rh._get_instance(cookiejar=rh.cookiejar)
+        called = False
+        original_close = session.close
+
+        def mock_close(*args, **kwargs):
+            nonlocal called
+            called = True
+            return original_close(*args, **kwargs)
+
+        monkeypatch.setattr(session, 'close', mock_close)
+
+        rh.close()
+
+        assert called
```

Comment:
formatting between this and the other one is inconsistent, but i would prefer
```suggestion
        monkeypatch.setattr(session, 'close', mock_close)
        rh.close()
        assert called
```